src/test/java/com/mongodb/jdbc/MongoMock.java

```diff
@@ -26,19 +28,31 @@
     @Mock protected static MongoCursor<MongoResultDoc> mongoCursor;
 
     @InjectMocks
-    protected static MongoConnection mongoConnection = new MongoConnection(uri, database, null);
+    protected static MongoConnection mongoConnection = new MySQLConnection(uri, database, null);
+
+    private static Field getDeclaredFieldFromClassOrSuperClass(Class c, String fieldName) throws NoSuchFieldException {
```

Comment:
What do you think of this for fixing the problem of getting a field from a super class? I made it generic even though we really only need it for `MongoConnection` and really only need to go up one level of the hierarchy tree.

Reply:
That's a good way to handle the problem of the field being in the class or any of the parent classes. 👍